examples/examples2d.jl

```diff
@@ -382,3 +382,64 @@
         render(A)
     end
 end
+
+@block AnshulSinghvi ["colors"] begin
+
+    @cell "Line with varying colors" [lines, colors, colorlegend] begin
+
+        using AbstractPlotting, # duh :)
+              ColorSchemes      # colormaps galore
+
+        AbstractPlotting.__init__();
+
+        t = range(0, stop=1, length=100) # times
+
+        θ = (6π) .* t    # angles
+
+        x = t .* cos.(θ) # x coords of spiral
+        y = t .* sin.(θ) # y coords of spiral
+
+        p1 = lines(
+            x,
+            y,
+            color = t,
+            colormap = ColorSchemes.magma.colors,
+            linewidth=3)
+
+        cm = colorlegend(
+            p1[end],           # access the plot of Scene p1
+            raw = true,        # without axes or grid
+            camera = campixel! # TODO describe
+            )
+
+        scene_final = vbox(p1, cm) # put the colorlegend and the plot together in a `vbox`
+
+    end
+
+    @cell "Viridis color legend" [colorlegend, colors] begin
+
+        using AbstractPlotting
+
+        AbstractPlotting.__init__();
+
+        scene = Scene()
+
+        cam2d!(scene) ## TODO what do this?
+
+        cmap = :viridis # default Colormap
+
+        c = to_colormap(cmap) # get colors of colormap
+
+        image!(        # to plot colors, an image is best
+            scene,
+            range(0, stop = 10, length = length(c)), # x range
+            range(0, stop = 1, length = length(c)),  # y range
+            reshape(c, (length(c),1)),               # reshape this to a matrix for the colors
+            show_axis = false                        # don't show axes
+        )
+
+        scene # show the scene
```

Comment:
I'd prefer:
```julia
 # default Colormap
c = to_colormap(:viridis) # get colors of colormap

image(        # to plot colors, an image is best
    0..10, # x range
    0..1,  # y range
    hcat(c, c), # reshape this to a matrix for the colors
    show_axis = false # don't show axes
)
```

Reply:
`camera = campixel!` is needed to have a concrete boundingbox in pixel units, so that the layouting in vbox can automatically give it the right size!

Reply:
We should try to just make campixel the default for legend... Probably not possible right away without some refactoring, though :( Will do this as part of nicely integrating the legend

Reply:
Makes sense, I'll change that.  Thanks!